bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -355,6 +356,7 @@ protected Point computeSize (int part, int state, GC gc, int wHint, int hHint) {
 							gc.setFont(gcFont);
 						}
 					}
+					width += getTextPadding(item, state);
```

Comment:
I would suggest to apply the padding two times to the overall width, so that the text is centered within the tab (as the padding is applied one time to the x offset of the text).

```suggestion
					width += getTextPadding(item, state) * 2;
```

Reply:
In that case, code is reverted to adding the the padding twice uniformly to the tab width. 
Also, in case close button is visible for unselected tab, the close button size is still added to tab width. If we remove the close button size, the text gets truncated and the padding would need to be increased.

With the goal to center the text within tab, close button size wouldn't matter much.